### PR TITLE
PLAT-469: test conveyor RestoreStep doesn't wake up SM

### PR DESCRIPTION
**Pull request by MikhailZhukov**
Test conveyor RestoreStep doesn't wake up SM

### CODE REVIEW

File: ledger-core/conveyor/smachine/slot_machine_test.go [hunk 1]
```diff
@@ -90,3 +90,86 @@ func TestSlotMachine_AddSMAndMigrate(t *testing.T) {
...
+	return ctx.Jump(s.stepWaitInfinity)
+}
+
+func (s *smTestRestoreStep) stepWaitInfinity(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	return ctx.Sleep().ThenRepeat()
+}
+
+func (s *smTestRestoreStep) migrate(ctx smachine.MigrationContext) smachine.StateUpdate {
+	s.wasMigrated = true
+	affectedStep := ctx.AffectedStep()
```
Comment (cyraxred) [inline ledger-core/conveyor/smachine/slot_machine_test.go:120]: add s.wasContinued = false here

File: ledger-core/conveyor/smachine/slot_machine_test.go [hunk 2]
```diff
@@ -90,3 +90,86 @@ func TestSlotMachine_AddSMAndMigrate(t *testing.T) {
...
+	affectedStep.Transition = func(ctx smachine.ExecutionContext) smachine.StateUpdate {
```
Comment (cyraxred) [inline ledger-core/conveyor/smachine/slot_machine_test.go:121]: remove it and replace with 
return ctx.RestoreStep(ctx.AffectedStep())

File: ledger-core/conveyor/smachine/slot_machine_test.go [hunk 3]
```diff
@@ -90,3 +90,86 @@ func TestSlotMachine_AddSMAndMigrate(t *testing.T) {
...
+	return s
+}
+
+func (s *smTestRestoreStep) stepInit(ctx smachine.InitializationContext) smachine.StateUpdate {
+	ctx.SetDefaultMigration(s.migrate)
+	return ctx.Jump(s.stepWaitInfinity)
+}
+
+func (s *smTestRestoreStep) stepWaitInfinity(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	return ctx.Sleep().ThenRepeat()
```
Comment (cyraxred) [inline ledger-core/conveyor/smachine/slot_machine_test.go:115]: add as first line
s.wasContinued = true

File: ledger-core/conveyor/smachine/slot_machine_test.go (hunk 2 above)
Comment (MikhailZhukov) [reply ledger-core/conveyor/smachine/slot_machine_test.go]: done

File: ledger-core/conveyor/smachine/slot_machine_test.go (hunk 3 above)
Comment (MikhailZhukov) [reply ledger-core/conveyor/smachine/slot_machine_test.go:116]: done

**Pull request merged**